lib/resque/plugins/lock.rb

```diff
@@ -75,6 +75,10 @@ def before_enqueue_lock(*args)
         now > Resque.redis.getset(key, timeout).to_i
       end
 
+      def before_perform_lock(*args)
+        Resque.redis.del(lock(*args)) if @unlock_while_performing
+      end
+
       def around_perform_lock(*args)
```

Comment:
The thinking being to avoid an unnecessary call to `del` if the lock is already gone?